### ISSUE TITLE
Question of Training Dataset

### ISSUE THREAD

**Issue by finallymint** [GeorgeVern/lmcor#4]
Dear Giorgos @GeorgeVern 
I have the same question with Equation (2), is the y in the training set obtained by selecting the best one from the candidate set C, or is it directly using the label y that comes with the benchmark?
<img width="393" alt="image" src="https://github.com/user-attachments/assets/d874bf46-156a-4afe-b135-9e4ebfa7b924">

**Comment by GeorgeVern** [comment]
Thank you for your interesting question!

In principle, our approach is designed to work with existing training sets so y would be the label. However, if you have a criterion or metric to select the best candidate as y, that would be very interesting, and it should work as well (similar to [Generating Sequences by Learning to Self-Correct](https://openreview.net/forum?id=hH36JeQZDaO) ). 

In this case, a follow-up question would be whether you should leave the best candidate candidate as input as well. This would encourage the model to focus on ranking, rather than editing and combining. Alternatively, you could sample N+1 candidates, select the best one based on your criterion, and treat the remaining N as input. I’d be interested to hear if this approach works for you!

**Comment by finallymint** [comment]
Thank you very much for answering my questions! This work is meaningful and your response really inspires me a lot! Thank you very much!
